packages/compass-logging/src/react.ts

```diff
@@ -0,0 +1,45 @@
+import createLoggerAndTelemetry from './logger';
+import type { LoggerAndTelemetry } from './logger';
+
+export function useLoggerAndTelemetry(
+  component: string,
+  React: { useRef: any }
+): LoggerAndTelemetry {
+  const loggerRef = React.useRef();
+  if (!loggerRef.current) {
+    loggerRef.current = createLoggerAndTelemetry(component);
+  }
+  return loggerRef.current as LoggerAndTelemetry;
+}
+
+export function useTrackOnChange(
+  component: string,
+  value: unknown,
```

Comment:
It might be a bit more obvious at the call sites what this is referring to if it were a dependencies array the same way `useEffect()` itself takes one, i.e. 

```suggestion
  value: unknown[],
```